README.md

```diff
@@ -1,13 +1,408 @@
+
+> In 24 hours you might be able to learn some of the syntax of C++ (if you already know another language), but you couldn't learn much about how to use the language. In short, if you were, say, a Basic programmer, you could learn to write programs in the style of Basic using C++ syntax, but you couldn't learn what C++ is actually good (and bad) for. So what's the point? Alan Perlis once said: "A language that doesn't affect the way you think about programming, is not worth knowing" -- [Peter Norvig][1]
+
+[1]:http://www.norvig.com/21-days.html
+
+This is the principle of linguistic relativity also known as the [Sapir-Whorf hypothesis][2] shows that the constraints of a language are what most often limit our thought processing. This idea, when applied to programming languages shows how Lisp programmers use macros for everything, Python programmers even in Ruby have code that looks like python.  
+
+[2]:http://en.wikipedia.org/wiki/Linguistic_relativity
+
+One of the key components to developing C++ is learnability or more importantly, how easy is it for users to accomplish basic tasks. 
+
+C++ Collections is a C++ 11 library that provides lazy evaluation and functional transformations of the data to lighten the burden of using functional operations such as map, range, and filter for the user. To use C++ Streams, download them from Github and simply *#include collections.h* in the header of your file after dragging it in.
+
+----
+## What is C++ Collections?
+C++ Collections provides abstractions on a set of ordered data. All examples below assume that we are `using namespace cpp_collections`.
+The Class collections is used to support functional style transformations as demonstrated above. Assume you would like to create a list of numbers from 1 to 100, sum them, and then store the value in an int. Functional operators allow these operations to happen in nearly one line of code.
+
+```cpp
+int sum = range(1,101).reduceLeft([](int x, int y) {return x + y;});
+```
+
+The syntax you will note is similar to both Haskell and Scala.
+
+```haskell
+sum = foldl1 (+) [1..100]
+```
+
+```scala
+val sum = Range(1, 100).reduce((a,b) => a+b)
+```
+
+The main abstraction introduced by this library is Collection. The capitalization om the word 'Collection' is important to distinguish it from collection. The Collection is different from standard collections in C++ in that:
+
+1. It is functional in nature. All operations on a Collection produce some form of result by modifying the source in memory. 
+2. It is concise. After introducing lambda functions in C++11, and functors long before that, C++ has emerged as a viable language in which to write programs that conform to the functional paradigm. There is no template overhead to calling functions.
+3. Laziness seeking. We use laziness to optimize many Collections operations. So calling slice on the first five consecutive numbers will not require all numbers in the list to be examined. 
```

Comment:
Yeah I had that in there - like 3 and 4 before I saw the Streams class - I was figuring we were working on it so I had it in place, but now that it's under the Streams class I'll split them off and list them separately.